From: Jeff Moyer <jmoyer@redhat.com>
Date: Wed, 22 Apr 2009 12:15:13 -0400
Subject: [s390] enable raw devices
Message-id: x49ljpsoeb2.fsf@segfault.boston.devel.redhat.com
O-Subject: [rhel5 patch] s390: enable raw devices
Bugzilla: 452534

Hi,

Bugzilla 452534 requests that we enable raw devices on s390x so that
Oracle can certify 10g RAC on that platform.  There was some discussion
of whether or not this is strictly necessary, as Oracle should have
support for opening the block device with O_DIRECT by now.  I don't
think it matters, we should just enable it.

I was only able to test binding and unbinding a raw device, as the RHTS
system reservation defaults to using up all block devices (so opening
the raw device returns EBUSY).  I'll see if we can get customer testing
for this.

Note that the driver is still marked OBSOLETE in Kconfig.  It is also
this way for the generic drivers/char/Kconfig, and our release notes
mention that this is no longer the case, so I think it's fine as-is.

Comments, as always, are appreciated.

Cheers,
Jeff

diff --git a/drivers/s390/Kconfig b/drivers/s390/Kconfig
index b9271e3..47e1d30 100644
--- a/drivers/s390/Kconfig
+++ b/drivers/s390/Kconfig
@@ -51,6 +51,26 @@ config UNIX98_PTY_COUNT
 	  When not in use, each additional set of 256 PTYs occupy
 	  approximately 8 KB of kernel memory on 32-bit architectures.
 
+config RAW_DRIVER
+       tristate "RAW driver (/dev/raw/rawN) (OBSOLETE)"
+       help
+        The raw driver permits block devices to be bound to /dev/raw/rawN.
+        Once bound, I/O against /dev/raw/rawN uses efficient zero-copy I/O.
+        See the raw(8) manpage for more details.
+
+        The raw driver is deprecated and may be removed from 2.7
+        kernels. Applications should simply open the device (eg /dev/hda1)
+        with the O_DIRECT flag.
+
+config MAX_RAW_DEVS
+       int "Maximum number of RAW devices to support (1-8192)"
+       depends on RAW_DRIVER
+       default "256"
+       help
+        The maximum number of RAW devices that are supported.
+        Default is 256. Increase this number in case you need lots of
+        raw devices.
+
 config HANGCHECK_TIMER
 	tristate "Hangcheck timer"
 	help
